From 7eb7d32fb166608444d0779517811554dd62069b Mon Sep 17 00:00:00 2001
From: Rohit Athavale <rohit.athavale@xilinx.com>
Date: Thu, 5 Apr 2018 10:20:32 -0700
Subject: [PATCH 0342/1851] dt-bindings: display: xlnx: mixer: Add supported
 formats table

commit 8b55c1c5252d976cb05f4dc2f8845388b866f502 from
https://github.com/Xilinx/linux-xlnx.git

This commit adds a list of supported formats that the driver supports.

Signed-off-by: Rohit Athavale <rohit.athavale@xilinx.com>
Reviewed-by: Saurabh Sengar <saurabhs@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/display/xlnx/xlnx,mixer.txt      | 27 ++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
index 30739c3d35c3..321f09385ef7 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,mixer.txt
@@ -18,7 +18,7 @@ Required properties:
  - xlnx,ppc: pixel per clock for mixer
  - layer_[x]: node for [x] layer
  - xlnx,layer-id: layer identifier number
- - xlnx,vformat: video format for layer
+ - xlnx,vformat: video format for layer. See list of supported formats below.
  - xlnx,layer-max-width: max layer width, mandatory for master layer
    for overlay layers if scaling is alowed then this is mandatory otherwise
    not required for overlay layers
@@ -37,6 +37,31 @@ Optional properties:
  - xlnx,layer-scale: denotes layer can be scale to 2x and 4x
  - xlnx,logo-layer: denotes logo layer is enable
  - logo: logo layer
+
+Supported Formats:
+	Mixer IP Format		Driver supported Format String
+	 BGR888			"BG24"
+	 RGB888			"RG24"
+	 XBGR2101010		"XB30"
+	 XRGB8888		"XR24"
+	 RGBA8888		"RA24"
+	 ABGR8888		"AB24"
+	 ARGB8888		"AR24"
+	 XBGR8888		"XB24"
+	 YUYV			"YUYV"
+	 UYVY			"UYVY"
+	 AYUV			"AYUV"
+	 NV12			"NV12"
+	 NV16			"NV16"
+	 Y8			"GREY"
+	 Y10			"Y10 " (Note: Space included)
+	 XVUY2101010		"XV30"
+	 VUY888			"VU24"
+	 XVUY8888		"XV24"
+	 XV15			"XV15"
+	 XV20			"XV20"
+Note : Format strings are case sensitive.
+
 Example:
 	v_mix_0: v_mix@80100000 {
 		compatible = "xlnx,mixer-3.0";
-- 
2.31.1

